hoodie-client/src/main/java/com/uber/hoodie/index/bloom/HoodieBloomIndex.java

```diff
@@ -300,57 +295,70 @@ public boolean isImplicitWithStorage() {
   }
 
   /**
-   * if we dont have key ranges, then also we need to compare against the file. no other choice if
-   * we do, then only compare the file if the record key falls in range.
+   * if we dont have key ranges, then also we need to compare against the file. no other choice if we do, then only
+   * compare the file if the record key falls in range.
    */
   boolean shouldCompareWithFile(BloomIndexFileInfo indexInfo, String recordKey) {
     return !indexInfo.hasKeyRanges() || indexInfo.isKeyInRange(recordKey);
   }
 
 
   /**
-   * For each incoming record, produce N output records, 1 each for each file against which the
-   * record's key needs to be checked. For datasets, where the keys have a definite insert order
-   * (e.g: timestamp as prefix), the number of files to be compared gets cut down a lot from range
-   * pruning.
+   * For each incoming record, produce N output records, 1 each for each file against which the record's key needs to be
+   * checked. For datasets, where the keys have a definite insert order (e.g: timestamp as prefix), the number of files
+   * to be compared gets cut down a lot from range pruning.
    *
-   * Sub-partition to ensure the records can be looked up against files & also prune
-   * file<=>record comparisons based on recordKey
-   * ranges in the index info.
+   * Sub-partition to ensure the records can be looked up against files & also prune file<=>record comparisons based on
+   * recordKey ranges in the index info.
    */
   @VisibleForTesting
   JavaPairRDD<String, Tuple2<String, HoodieKey>> explodeRecordRDDWithFileComparisons(
       final Map<String, List<BloomIndexFileInfo>> partitionToFileIndexInfo,
       JavaPairRDD<String, String> partitionRecordKeyPairRDD) {
+    Map<String, IndexLookUpTree> partitionToFileIndexLookUpTree = new HashMap<>();
+    Set<String> filesWithNoRanges = new HashSet<>();
```

Comment:
Can we introduce an interface and just get only one of these structures built out based on whether range pruning is on or off?

Reply:
+1

Reply:
@nsivabalan : filesWithNoRanges should also be a map keyed by partition-path. right ? Otherwise, won't it result in unnecessary matches for files (with no range info) even if the partition-path is different ?

Reply:
` IndexFileFilter` has been introduced